pkg/gadgets/trace/exec/tracer/bpf/execsnoop.bpf.c

```diff
@@ -166,44 +163,34 @@ static __always_inline bool has_upper_layer(struct inode *inode)
 	return upperdentry != NULL;
 }
 
-SEC("tracepoint/syscalls/sys_exit_execve")
-int ig_execve_x(struct syscall_trace_exit *ctx)
+// tracepoint/sched/sched_process_exec is called after a successful execve
+SEC("tracepoint/sched/sched_process_exec")
+int ig_sched_exec(struct trace_event_raw_sched_process_exec *ctx)
 {
-	pid_t execs_lookup_key;
-	int ret;
+	u32 execs_lookup_key = ctx->old_pid;
 	struct event *event;
 	struct task_struct *task = (struct task_struct *)bpf_get_current_task();
 	struct task_struct *parent = BPF_CORE_READ(task, real_parent);
 	struct file *exe_file;
 	char *exepath;
 
-	ret = ctx->ret;
-	execs_lookup_key = gadget_get_exec_caller_pid(ret);
-	if (!execs_lookup_key)
-		return 0;
-
 	event = bpf_map_lookup_elem(&execs, &execs_lookup_key);
-
 	if (!event)
 		return 0;
-	if (ignore_failed && ret < 0)
-		goto cleanup;
 
-	if (ret == 0) {
-		struct inode *inode =
-			BPF_CORE_READ(task, mm, exe_file, f_inode);
-		if (inode) {
-			event->upper_layer = has_upper_layer(inode);
-		}
-
-		struct inode *pinode =
-			BPF_CORE_READ(parent, mm, exe_file, f_inode);
-		if (pinode) {
-			event->pupper_layer = has_upper_layer(pinode);
-		}
+	bpf_map_delete_elem(&execs, &execs_lookup_key);
```

Comment:
I thought it was safe because some other ebpf programs in the Linux tree have this pattern with the comment that the value are still accessible under rcu:
https://github.com/torvalds/linux/blob/933069701c1b507825b514317d4edd5d3fd9d417/tools/testing/selftests/bpf/progs/bpf_iter_bpf_hash_map.c#L66-L71

There are similar explanations at:
https://github.com/xdp-project/xdp-tutorial/issues/203

However, this thread seems to say something else (I am not sure to understand it):
https://lkml.org/lkml/2018/10/3/744

I find the code easier to read when bpf_map_delete_elem is done early, so we don't have to remember to delete the entry for each `return;`. What do you think?

Reply:
I personally prefer to have the delete operation at the end, it's a bit difficult to understand that you can continue to use a value after is has been deleted. 

In any case, the PR LGTM, I'll approve it and it's up to you which approach to use.

Reply:
I moved the bpf_map_delete_elem at the end. I will merge once the CI completes.